src/hipscat/io/paths.py

```diff
@@ -104,24 +110,57 @@ def pixel_catalog_files(
         catalog_base_dir (FilePointer): base directory of the catalog (includes catalog name)
         pixels (List[HealpixPixel]): the healpix pixels to create pointers to
         storage_options (dict): the storage options for the file system to target when generating the paths
+        query_params (dict): query parameters to append to the url for http file systems.
+            An example of the expected `query_params` dictionary format:
+            {
+                'columns': ['RA', 'DEC', 'r_auto', 'g_auto'],
+                'filters': ['r_auto>=10', 'g_auto<18']
+            }
+
     Returns (List[FilePointer]):
         A list of paths to the pixels, in the same order as the input pixel list.
     """
     fs, _ = get_fs(catalog_base_dir, storage_options)
     base_path_stripped = catalog_base_dir.removesuffix(fs.sep)
+
+    url_params = ""
+    if isinstance(fs, HTTPFileSystem) and query_params:
+        url_params = dict_to_query_urlparams(query_params)
+
     return [
         fs.sep.join(
             [
                 base_path_stripped,
                 f"{ORDER_DIRECTORY_PREFIX}={pixel.order}",
                 f"{DIR_DIRECTORY_PREFIX}={pixel.dir}",
-                f"{PIXEL_DIRECTORY_PREFIX}={pixel.pixel}.parquet",
+                f"{PIXEL_DIRECTORY_PREFIX}={pixel.pixel}.parquet" + url_params,
             ]
         )
         for pixel in pixels
     ]
 
 
+def dict_to_query_urlparams(query_params: dict) -> str:
```

Comment:
Can you add some unit tests specific to this method? e.g. for null/empty values in the dict, characters that will need to be escaped in the resulting URL segment.

Reply:
Refactored this method, 
Thought it was the best way to do it, but I was just complicating things.

Also added the tests!

Reply:
Yes - this is much cleaner!